packages/desktop/views/dashboard/account-management/AccountManagement.svelte

```diff
@@ -1,6 +1,50 @@
 <script lang="ts">
+    import { truncateString } from '@core/utils'
     import { selectedWallet } from '@core/wallet/stores'
-    import { Height, Pane } from '@ui'
+    import { AccountAddress, AccountOutput, ImplicitAccountCreationAddress, OutputData } from '@iota/sdk/out/types'
+    import { Height, Pane, TextType, Text, ClickableTile, FontWeight, Pill } from '@ui'
+    import { onMount } from 'svelte'
+    import { localize } from '@core/i18n'
+    import { getBech32AddressFromAddressTypes } from '@core/wallet/utils'
+    import features from '@features/features'
+
+    let allAccounts: OutputData[] = []
+    let accounts: OutputData[] = []
+    let implicitAccounts: OutputData[] = []
+
+    async function getAccountData() {
+        accounts = await $selectedWallet.accounts()
+        implicitAccounts = await $selectedWallet.implicitAccounts()
+        allAccounts = [...accounts, ...implicitAccounts]
+    }
+
+    function isAnAccount(output: OutputData) {
+        return accounts.find((account) => account.outputId === output.outputId)
+    }
+
+    function isAnImplicitAccount(output: OutputData) {
+        return implicitAccounts.find((account) => account.outputId === output.outputId)
+    }
+
+    function formatAndTruncateAccount(output) {
+        let address: string
+        if ((output as AccountOutput).accountId) {
+            address = getBech32AddressFromAddressTypes(new AccountAddress((output as AccountOutput).accountId))
+        } else {
+            address = getBech32AddressFromAddressTypes(
+                new ImplicitAccountCreationAddress(output.unlockConditions[0].address.pubKeyHash).address()
+            )
```

Comment:
We can improve this
```suggestion
            const implicitUnlockCondition = (
                (output as CommonOutput).unlockConditions.find(
                    (cmnOutput) => cmnOutput.type === UnlockConditionType.Address
                ) as AddressUnlockCondition
            )
            if (implicitUnlockCondition.address.type === AddressType.ImplicitAccountCreation) {
                address = getBech32AddressFromAddressTypes(
                    new ImplicitAccountCreationAddress(output.unlockConditions[0].address.pubKeyHash).address()
                )   
            }```

we can externalize the function of `isImplicitAccount` and use it here and in `handleNewOutput`